includes/class-wc-cart-session.php

```diff
@@ -130,10 +130,9 @@ public function get_cart_from_session() {
 
 			} elseif ( ! $product->is_purchasable() ) {
 				$update_cart_session = true;
-				/* translators: %s: product name */
-				wc_add_notice( sprintf( __( '%s has been removed from your cart because it can no longer be purchased. Please contact us if you need assistance.', 'woocommerce' ), $product->get_name() ), 'error' );
-				do_action( 'woocommerce_remove_cart_item_from_session', $key, $values );
-
+				/* translators: %1$s: product name. %2$s reason why */
+				wc_add_notice( sprintf( __( '%s has been removed from your cart because it can no longer be purchased. Please contact us if you need assistance.', 'woocommerce' ), $product->get_name() ), 'error' );				do_action( 'woocommerce_remove_cart_item_from_session', $key, $values );
```

Comment:
Some reason to change the `do_action()` from that line?

Reply:
That was an accident.